segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/StreamSegmentStore.java

```diff
@@ -27,43 +27,51 @@
+     * byte range will be appended as a contiguous block. This method guarantees ordering (among
+     * subsequent calls).
      *
      * @param streamSegmentName The name of the StreamSegment to append to.
-     * @param offset            The offset at which to append. If the current length of the StreamSegment does not equal
-     *                          this value, the operation will fail with a BadOffsetException.
-     * @param data              The data to add.
-     * @param attributeUpdates  A Collection of Attribute-Values to set or update. May be null (which indicates no updates).
-     *                          See Notes about AttributeUpdates in the interface Javadoc.
-     * @param timeout           Timeout for the operation
-     * @return A CompletableFuture that, when completed normally, will indicate the append completed successfully.
-     * If the operation failed, the future will be failed with the causing exception.
-     * @throws NullPointerException     If any of the arguments are null, except attributeUpdates.
-     * @throws IllegalArgumentException If the StreamSegment Name is invalid (NOTE: this doesn't check if the StreamSegment
-     *                                  does not exist - that exception will be set in the returned CompletableFuture).
+     * @param offset The offset at which to append. If the current length of the StreamSegment does
+     *            not equal this value, the operation will fail with a BadOffsetException.
+     * @param data The data to add.
+     * @param attributeUpdates A Collection of Attribute-Values to set or update. May be null (which
+     *            indicates no updates). See Notes about AttributeUpdates in the interface Javadoc.
+     * @param timeout Timeout for the operation
+     * @return A CompletableFuture that, when completed normally, will indicate the append completed
+     *         successfully and contains the new length of the segment. If the operation failed, the
+     *         future will be failed with the causing exception. (NOTE: the length is not
+     *         necessarily the same as offset immediately following the data because the append may
+     *         have been batched together with others internally.)
+     * @throws NullPointerException If any of the arguments are null, except attributeUpdates.
+     * @throws IllegalArgumentException If the StreamSegment Name is invalid (NOTE: this doesn't
+     *             check if the StreamSegment does not exist - that exception will be set in the
+     *             returned CompletableFuture).
      */
-    CompletableFuture<Void> append(String streamSegmentName, long offset, byte[] data, Collection<AttributeUpdate> attributeUpdates, Duration timeout);
+    CompletableFuture<Long> append(String streamSegmentName, long offset, byte[] data, Collection<AttributeUpdate> attributeUpdates, Duration timeout);
```

Comment:
Technically append-with-offset doesn't need to return this (as it should be the same as `offset`. Did you put this here just for consistency?

Reply:
Yes. They get handled the same in the caller, so it was easier for them to work the same way.